### PR TITLE
Fix issue #185

### PR DESCRIPTION
**Pull request by OzancanKaratas**
Hello @r10s, may you test this changes? If you run a problem, please visit my fork and continue to change.

### CODE REVIEW

**Comment by OzancanKaratas** [comment]
This marks is suitable for a .strings file. I've leaved out 2 backslashes in a line. Because it displays single backslash. I removed restrictions in my fork for more.

File: tools/convertTranslations.js [hunk 1]
```diff
@@ -21,15 +21,15 @@ function parseAndroid(data) {
...
-      replace(/\\t/g, '\t').
-      replace(/\\r/g, '\r').
-      replace(/\\n/g, '\n').
-      replace(/\\\\/g, '\\').
-      replace(/\$s/ig, '$@')])
+      replace(/\t/g, '\t').
+      replace(/\r/g, '\r').
+      replace(/\n/g, '\n').
+      replace(/\\/g, '\\').
+      replace(/$s/ig, '$@')])
```
Comment (r10s) [inline tools/convertTranslations.js]: this looks wrong to me. `$` matches end-of-line. i think the escaping is needed here.

**Comment by r10s** [comment]
> If you run a problem, please visit my fork and continue to change.

tried that, however, the invitation is outdated?

**Comment by OzancanKaratas** [comment]
Please retry. Also my fork does not restricted.

**Comment by OzancanKaratas** [comment]
I found several typos, but I don't know change them in untranslated strings. Source files does not contain the strings. How do I change the strings?

File: tools/convertTranslations.js [hunk 2]
```diff
@@ -21,15 +21,15 @@ function parseAndroid(data) {
...
-      replace(/\\t/g, '\t').
-      replace(/\\r/g, '\r').
-      replace(/\\n/g, '\n').
-      replace(/\\\\/g, '\\').
-      replace(/\$s/ig, '$@')])
+      replace(/\t/g, '\t').
+      replace(/\r/g, '\r').
+      replace(/\n/g, '\n').
+      replace(/\\/g, '\\').
+      replace(/$s/ig, '%@')])
```
Comment (r10s) [inline tools/convertTranslations.js]: this is still wrong, the original statement was correct here, `replace(/\$s/ig, '$@')])`

**Comment by r10s** [comment]
> I found several typos, but I don't know change them in untranslated strings. Source files does not contain the strings. How do I change the strings?

most strings in deltachat-ios repo are generated

- translations are edited at https://www.transifex.com/delta-chat/delta-chat-app/ (you need a free transifex account for that)
- the english source strings can be edited via a pr to https://github.com/deltachat/deltachat-android/blob/master/res/values/strings.xml

**Comment by OzancanKaratas** [comment]
I edited $s mark as %@. Please fetch the changes and retry. Or add me in main repository. I think this pull request should be merged a branch in the main repo. Because we can work more comfortable there.

**Comment by OzancanKaratas** [comment]
Final! Latest commit 076832f will fix the compile error.

**Comment by OzancanKaratas** [comment]
This can now be merged and closes #185.

File: tools/convertTranslations.js [hunk 3]
```diff
@@ -189,7 +189,7 @@ function toLocalizableStrings(lines) {
       if (key.startsWith("INFOPLIST.")) {
         continue;
       }
-      let value = line[1].replace(/\\/g, '\\\\').replace(/\n/g, '\\n').replace(/\r/g, '\\r').replace(/\t/g, '\\t').replace(/"/g, '\\"');
+      let value = line[1].replace(/\\/g, '\\').replace(/\n/g, '\n').replace(/\r/g, '\r').replace(/\t/g, '\t').replace(/"/g, '\"');
```
Comment (cyBerta) [inline tools/convertTranslations.js]: terms like `replace(/\\/g, '\\')` or `replace(/\n/g, '\n')` have no effect. `\\` will be replaced by `\\` and `\n` by `\n`.
That's why this line can be reduced to `let value = line[1];` You can also substitute `value` by `line[1]` on line 193

File: tools/convertTranslations.js [hunk 4]
```diff
@@ -164,7 +164,7 @@ function toInfoPlistStrings(lines) {
           continue;
         }
         key = key.replace('INFOPLIST.', '').replace(/\./gi, ' ').replace(/\_/gi, '-');
-        let value = line[1].replace(/\\/g, '\\\\').replace(/\n/g, '\\n').replace(/\r/g, '\\r').replace(/\t/g, '\\t').replace(/"/g, '\\"');
+        let value = line[1].replace(/\\/g, '\\').replace(/\n/g, '\n').replace(/\r/g, '\r').replace(/\t/g, '\t').replace(/"/g, '\"');
```
Comment (cyBerta) [inline tools/convertTranslations.js]: same issue like in line 192

File: tools/convertTranslations.js (hunk 3 above)
Comment (cyBerta) [reply tools/convertTranslations.js]: To be clear: the original code `replace(/\\/g, '\\\\')` was wrong! So good catch here!

File: tools/convertTranslations.js [hunk 5]
```diff
@@ -21,14 +21,14 @@ function parseAndroid(data) {
...
       replace(/&gt;/g, '>').
       replace(/&amp;/g, '&').
-      replace(/\\t/g, '\t').
-      replace(/\\r/g, '\r').
-      replace(/\\n/g, '\n').
-      replace(/\\\\/g, '\\').
+      replace(/\t/g, '\t').
+      replace(/\r/g, '\r').
+      replace(/\n/g, '\n').
+      replace(/\\/g, '\\').
```
Comment (cyBerta) [inline tools/convertTranslations.js]: again. this code has no effect. it replaces the regex with itself. You can remove it completely

**Comment by OzancanKaratas** [comment]
I edited the code by strings files. If there are something I forget, please come my fork.

**Comment by cyBerta** [comment]
@OzancanKaratas What do you mean by that? I checked out your feature branch when I was reviewing the merge request. I also told you what to change. Of course I can do it by my own, but I thought it would be nice for you if you're in the commit history and the contributers list.

**Comment by OzancanKaratas** [comment]
@r10s reviewed this pull request and he didn't found a problem more than one. I fixed the problem. If I could not fixed, please do it in my unrestricted fork. At this time, I don't know what should I do more.

**Comment by cyBerta** [comment]
ok, no problem. I'll do that

**Comment by r10s** [comment]
@cyBerta cool, thanks

**Comment by cyBerta** [comment]
I continued the work on the branch #234 . All of your commits are included there @OzancanKaratas

**Pull request closed without merging**